lychee-lib/src/client.rs

```diff
@@ -456,21 +496,22 @@ impl Client {
         }
 
         let mut retries: u64 = 0;
-        let mut wait = self.retry_wait_time;
+        let mut wait_time = self.retry_wait_time;
 
         let mut status = self.check_default(uri).await;
         while retries < self.max_retries {
             if status.is_success() {
                 return status;
             }
-            sleep(wait).await;
+            sleep(wait_time).await;
             retries += 1;
-            wait *= 2;
+            wait_time = wait_time.saturating_mul(2);
             status = self.check_default(uri).await;
         }
 
         // Pull out the heavy machinery in case of a failed normal request.
         // This could be a GitHub URL and we ran into the rate limiter.
+        // TODO: We should first try to parse the URI as GitHub URI first (Lucius, Jan 2023)
```

Comment:
Why is that? We should mention the reason here.
It's certainly a common use-case to check Github URLs, but only because many people use lychee as a Github action. Then again the additional parsing is probably negligible, so feel free to move it up.

Reply:
ideally we can treat GitHub urls as others (suppose there is no rate limit).

we need octocrab client to deal with GitHub urls only when we hit the rate limit. but in this case the rate limit is not going away at time soon, so any subsequent GitHub urls always results in at least two checks.

Reply:
So you mean the original request to a Github URL using `reqwest` also counts into the Github rate limit, which is why it makes no sense to check with `reqwest` first and only then use `octocrab`, right? I guess I understand now.
Yeah you might be right. But in that case, why not move the Github URL check up in this pull request? Wouldn't that immediately resolve the `TODO`? 🤔

Reply:
Guess we can also do that later at your preference.

Reply:
> So you mean the original request to a Github URL using reqwest also counts into the Github rate limit, which is why it makes no sense to check with reqwest first and only then use octocrab, right? I guess I understand now.

Exactly.

> why not move the Github URL check up in this pull request?

The initial intent of this PR is to improve ONLY the documentation.
Specifically, we have not explained the error scenarios very well before.

The by-product is, during this process I found some errors never happens.